### DISCUSSION TITLE
No internet connection?

### DISCUSSION THREAD

**Discussion by Mercury-Fred** [danielaparker/jsoncons#537] [Q&A]
Hi all,

Is there a suggested approach to use the schema validation w/o an internet connection?  Typically the schema references the base schema via URL, e.g. https://json-schema.org/draft/2020-12/schema.  Is it a manual process to download all the resources referenced by the URL and convert them to static files in my implementation?  Or some off the shelf solution?  Thanks!

**Comment by danielaparker** [comment]
You need to supply a URIResolver to resolve references to resources identified by URI's, see [Using a URIResolver to resolve references to schemas defined in external files](https://github.com/danielaparker/jsoncons/blob/master/doc/ref/jsonschema/jsonschema.md#eg3)
